app/src/main/java/util/extensions/FirebaseFirestoreExt.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2014-2019 Julien Guerinet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.guerinet.mymartlet.util.extensions
+
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.suspendCancellableCoroutine
+import timber.log.Timber
+import kotlin.coroutines.resume
+
+/**
+ * Extensions for the FirebaseFirestore
+ * @author Julien Guerinet
+ * @since 2.0.0
+ */
+
+/**
+ * Retrieves the collection with [collectionName] from the Firestore, parses it using the [init] function, and returns
+ *  the corresponding list
+ */
+suspend inline fun <reified T : Any> FirebaseFirestore.get(
```

Comment:
I don't know if this implementation is right. I'm essentially trying to convert a callback based system to coroutines. The only "concern" I have with this at this point is that I am not specifying that this should run on `io`, which means that you need to remember to set the right context when calling this. But I couldn't figure out how to do that properly.

Reply:
Just wrap everything with `withContext(Dispatchers.IO)`